### ISSUE TITLE
Worse  performance  with multiple GPU  training with Keras

### ISSUE THREAD

**Issue by lyw615** [keras-team/keras#14573]
The  network efficient net B3 used for Image Classification get worse performance  with multiple GPU  training.   The single GPU training get valid dataset accuracy at 0.94.  After multiple training,  the valid dataset accuracy fluctuates  between  0.49 to 0.53 .And  the loss value and accuracy of training dataset  are hard to change with bad metrics value. Just like the model  is hard to learn from training data and gives a  classfication result randomly . But the GPU are  still running for the network.  May the optimizer run the problem?
` 6/650 [..............................] - ETA: 35s - loss: 10.7454 - acc: 0.3333 - categorical_crossentropy: 10.7454
 12/650 [..............................] - ETA: 34s - loss: 10.7454 - acc: 0.3333 - categorical_crossentropy: 10.7454
 18/650 [..............................] - ETA: 33s - loss: 11.6408 - acc: 0.2778 - categorical_crossentropy: 11.6408
 24/650 [>.............................] - ETA: 33s - loss: 11.4170 - acc: 0.2917 - categorical_crossentropy: 11.4170
 30/650 [>.............................] - ETA: 32s - loss: 10.2081 - acc: 0.3667 - categorical_crossentropy: 10.2081
 36/650 [>.............................] - ETA: 32s - loss: 8.9545 - acc: 0.4444 - categorical_crossentropy: 8.9545  
 42/650 [>.............................] - ETA: 31s - loss: 8.8266 - acc: 0.4524 - categorical_crossentropy: 8.8266
 48/650 [=>............................] - ETA: 31s - loss: 8.3948 - acc: 0.4792 - categorical_crossentropy: 8.3948
 54/650 [=>............................] - ETA: 31s - loss: 8.6560 - acc: 0.4630 - categorical_crossentropy: 8.6560
 60/650 [=>............................] - ETA: 30s - loss: 8.5963 - acc: 0.4667 - categorical_crossentropy: 8.5963
 66/650 [==>...........................] - ETA: 30s - loss: 8.3033 - acc: 0.4848 - categorical_crossentropy: 8.3033
 72/650 [==>...........................] - ETA: 30s - loss: 8.2829 - acc: 0.4861 - categorical_crossentropy: 8.2829
 78/650 [==>...........................] - ETA: 29s - loss: 8.2657 - acc: 0.4872 - categorical_crossentropy: 8.2657
 84/650 [==>...........................] - ETA: 29s - loss: 8.2509 - acc: 0.4881 - categorical_crossentropy: 8.2509
 90/650 [===>..........................] - ETA: 29s - loss: 8.4172 - acc: 0.4778 - categorical_crossentropy: 8.4172
 96/650 [===>..........................] - ETA: 28s - loss: 8.3948 - acc: 0.4792 - categorical_crossentropy: 8.3948
102/650 [===>..........................] - ETA: 28s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
108/650 [===>..........................] - ETA: 28s - loss: 7.9098 - acc: 0.5093 - categorical_crossentropy: 7.9098
114/650 [====>.........................] - ETA: 27s - loss: 7.9177 - acc: 0.5088 - categorical_crossentropy: 7.9177
120/650 [====>.........................] - ETA: 27s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
126/650 [====>.........................] - ETA: 27s - loss: 7.9311 - acc: 0.5079 - categorical_crossentropy: 7.9311
132/650 [=====>........................] - ETA: 26s - loss: 7.9369 - acc: 0.5076 - categorical_crossentropy: 7.9369
138/650 [=====>........................] - ETA: 26s - loss: 8.2926 - acc: 0.4855 - categorical_crossentropy: 8.2926
144/650 [=====>........................] - ETA: 26s - loss: 8.2829 - acc: 0.4861 - categorical_crossentropy: 8.2829
150/650 [=====>........................] - ETA: 26s - loss: 8.1665 - acc: 0.4933 - categorical_crossentropy: 8.1665
156/650 [======>.......................] - ETA: 25s - loss: 8.2657 - acc: 0.4872 - categorical_crossentropy: 8.2657
162/650 [======>.......................] - ETA: 25s - loss: 8.3575 - acc: 0.4815 - categorical_crossentropy: 8.3575
168/650 [======>.......................] - ETA: 25s - loss: 8.3469 - acc: 0.4821 - categorical_crossentropy: 8.3469
174/650 [=======>......................] - ETA: 24s - loss: 8.4296 - acc: 0.4770 - categorical_crossentropy: 8.4296
180/650 [=======>......................] - ETA: 24s - loss: 8.6859 - acc: 0.4611 - categorical_crossentropy: 8.6859
186/650 [=======>......................] - ETA: 24s - loss: 8.6656 - acc: 0.4624 - categorical_crossentropy: 8.6656
192/650 [=======>......................] - ETA: 23s - loss: 8.6467 - acc: 0.4635 - categorical_crossentropy: 8.6467
198/650 [========>.....................] - ETA: 23s - loss: 8.4661 - acc: 0.4747 - categorical_crossentropy: 8.4661
204/650 [========>.....................] - ETA: 23s - loss: 8.2961 - acc: 0.4853 - categorical_crossentropy: 8.2961
210/650 [========>.....................] - ETA: 23s - loss: 8.3661 - acc: 0.4810 - categorical_crossentropy: 8.3661
216/650 [========>.....................] - ETA: 22s - loss: 8.3575 - acc: 0.4815 - categorical_crossentropy: 8.3575
222/650 [=========>....................] - ETA: 22s - loss: 8.4221 - acc: 0.4775 - categorical_crossentropy: 8.4221
228/650 [=========>....................] - ETA: 22s - loss: 8.3418 - acc: 0.4825 - categorical_crossentropy: 8.3418
234/650 [=========>....................] - ETA: 21s - loss: 8.3346 - acc: 0.4829 - categorical_crossentropy: 8.3346
240/650 [==========>...................] - ETA: 21s - loss: 8.2605 - acc: 0.4875 - categorical_crossentropy: 8.2605
246/650 [==========>...................] - ETA: 21s - loss: 8.1901 - acc: 0.4919 - categorical_crossentropy: 8.1901
252/650 [==========>...................] - ETA: 20s - loss: 8.1870 - acc: 0.4921 - categorical_crossentropy: 8.1870
258/650 [==========>...................] - ETA: 20s - loss: 8.3089 - acc: 0.4845 - categorical_crossentropy: 8.3089
264/650 [===========>..................] - ETA: 20s - loss: 8.4254 - acc: 0.4773 - categorical_crossentropy: 8.4254
270/650 [===========>..................] - ETA: 19s - loss: 8.4172 - acc: 0.4778 - categorical_crossentropy: 8.4172
276/650 [===========>..................] - ETA: 19s - loss: 8.4678 - acc: 0.4746 - categorical_crossentropy: 8.4678
282/650 [============>.................] - ETA: 19s - loss: 8.4591 - acc: 0.4752 - categorical_crossentropy: 8.4591
288/650 [============>.................] - ETA: 18s - loss: 8.4508 - acc: 0.4757 - categorical_crossentropy: 8.4508
294/650 [============>.................] - ETA: 18s - loss: 8.3880 - acc: 0.4796 - categorical_crossentropy: 8.3880
300/650 [============>.................] - ETA: 18s - loss: 8.3277 - acc: 0.4833 - categorical_crossentropy: 8.3277
306/650 [=============>................] - ETA: 18s - loss: 8.3224 - acc: 0.4837 - categorical_crossentropy: 8.3224
312/650 [=============>................] - ETA: 17s - loss: 8.2140 - acc: 0.4904 - categorical_crossentropy: 8.2140
318/650 [=============>................] - ETA: 17s - loss: 8.3125 - acc: 0.4843 - categorical_crossentropy: 8.3125
324/650 [=============>................] - ETA: 17s - loss: 8.3078 - acc: 0.4846 - categorical_crossentropy: 8.3078
330/650 [==============>...............] - ETA: 16s - loss: 8.3033 - acc: 0.4848 - categorical_crossentropy: 8.3033
336/650 [==============>...............] - ETA: 16s - loss: 8.3469 - acc: 0.4821 - categorical_crossentropy: 8.3469
342/650 [==============>...............] - ETA: 16s - loss: 8.3418 - acc: 0.4825 - categorical_crossentropy: 8.3418
348/650 [===============>..............] - ETA: 15s - loss: 8.3833 - acc: 0.4799 - categorical_crossentropy: 8.3833
354/650 [===============>..............] - ETA: 15s - loss: 8.3778 - acc: 0.4802 - categorical_crossentropy: 8.3778
360/650 [===============>..............] - ETA: 15s - loss: 8.3277 - acc: 0.4833 - categorical_crossentropy: 8.3277
366/650 [===============>..............] - ETA: 14s - loss: 8.3233 - acc: 0.4836 - categorical_crossentropy: 8.3233
372/650 [================>.............] - ETA: 14s - loss: 8.2757 - acc: 0.4866 - categorical_crossentropy: 8.2757
378/650 [================>.............] - ETA: 14s - loss: 8.1870 - acc: 0.4921 - categorical_crossentropy: 8.1870
384/650 [================>.............] - ETA: 13s - loss: 8.1430 - acc: 0.4948 - categorical_crossentropy: 8.1430
390/650 [=================>............] - ETA: 13s - loss: 8.1417 - acc: 0.4949 - categorical_crossentropy: 8.1417
396/650 [=================>............] - ETA: 13s - loss: 8.1405 - acc: 0.4949 - categorical_crossentropy: 8.1405
402/650 [=================>............] - ETA: 12s - loss: 8.1392 - acc: 0.4950 - categorical_crossentropy: 8.1392
408/650 [=================>............] - ETA: 12s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
414/650 [==================>...........] - ETA: 12s - loss: 7.9812 - acc: 0.5048 - categorical_crossentropy: 7.9812
420/650 [==================>...........] - ETA: 12s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
426/650 [==================>...........] - ETA: 11s - loss: 8.1347 - acc: 0.4953 - categorical_crossentropy: 8.1347
432/650 [==================>...........] - ETA: 11s - loss: 8.0217 - acc: 0.5023 - categorical_crossentropy: 8.0217
438/650 [===================>..........] - ETA: 11s - loss: 8.0222 - acc: 0.5023 - categorical_crossentropy: 8.0222
444/650 [===================>..........] - ETA: 10s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
450/650 [===================>..........] - ETA: 10s - loss: 8.1307 - acc: 0.4956 - categorical_crossentropy: 8.1307
456/650 [====================>.........] - ETA: 10s - loss: 8.0944 - acc: 0.4978 - categorical_crossentropy: 8.0944
462/650 [====================>.........] - ETA: 9s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590 
468/650 [====================>.........] - ETA: 9s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
474/650 [====================>.........] - ETA: 9s - loss: 8.0931 - acc: 0.4979 - categorical_crossentropy: 8.0931
480/650 [=====================>........] - ETA: 8s - loss: 8.0926 - acc: 0.4979 - categorical_crossentropy: 8.0926
486/650 [=====================>........] - ETA: 8s - loss: 8.0922 - acc: 0.4979 - categorical_crossentropy: 8.0922
492/650 [=====================>........] - ETA: 8s - loss: 8.1246 - acc: 0.4959 - categorical_crossentropy: 8.1246
498/650 [=====================>........] - ETA: 7s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
504/650 [======================>.......] - ETA: 7s - loss: 8.0910 - acc: 0.4980 - categorical_crossentropy: 8.0910
510/650 [======================>.......] - ETA: 7s - loss: 8.0274 - acc: 0.5020 - categorical_crossentropy: 8.0274
516/650 [======================>.......] - ETA: 6s - loss: 7.9966 - acc: 0.5039 - categorical_crossentropy: 7.9966
522/650 [=======================>......] - ETA: 6s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
528/650 [=======================>......] - ETA: 6s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
534/650 [=======================>......] - ETA: 6s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
540/650 [=======================>......] - ETA: 5s - loss: 7.9994 - acc: 0.5037 - categorical_crossentropy: 7.9994
546/650 [========================>.....] - ETA: 5s - loss: 8.0295 - acc: 0.5018 - categorical_crossentropy: 8.0295
552/650 [========================>.....] - ETA: 5s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
558/650 [========================>.....] - ETA: 4s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
564/650 [=========================>....] - ETA: 4s - loss: 8.0305 - acc: 0.5018 - categorical_crossentropy: 8.0305
570/650 [=========================>....] - ETA: 4s - loss: 7.9742 - acc: 0.5053 - categorical_crossentropy: 7.9742
576/650 [=========================>....] - ETA: 3s - loss: 7.9471 - acc: 0.5069 - categorical_crossentropy: 7.9471
582/650 [=========================>....] - ETA: 3s - loss: 7.9483 - acc: 0.5069 - categorical_crossentropy: 7.9483
588/650 [==========================>...] - ETA: 3s - loss: 7.9768 - acc: 0.5051 - categorical_crossentropy: 7.9768
594/650 [==========================>...] - ETA: 2s - loss: 8.0048 - acc: 0.5034 - categorical_crossentropy: 8.0048
600/650 [==========================>...] - ETA: 2s - loss: 8.0053 - acc: 0.5033 - categorical_crossentropy: 8.0053
606/650 [==========================>...] - ETA: 2s - loss: 7.9793 - acc: 0.5050 - categorical_crossentropy: 7.9793
612/650 [===========================>..] - ETA: 1s - loss: 7.9800 - acc: 0.5049 - categorical_crossentropy: 7.9800
618/650 [===========================>..] - ETA: 1s - loss: 7.9808 - acc: 0.5049 - categorical_crossentropy: 7.9808
624/650 [===========================>..] - ETA: 1s - loss: 7.9816 - acc: 0.5048 - categorical_crossentropy: 7.9816
630/650 [============================>.] - ETA: 1s - loss: 8.0335 - acc: 0.5016 - categorical_crossentropy: 8.0335
636/650 [============================>.] - ETA: 0s - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590
642/650 [============================>.] - ETA: 0s - loss: 8.0088 - acc: 0.5031 - categorical_crossentropy: 8.0088
648/650 [============================>.] - ETA: 0s - loss: 8.0342 - acc: 0.5015 - categorical_crossentropy: 8.0342
650/650 [==============================] - 39s 61ms/step - loss: 8.0590 - acc: 0.5000 - categorical_crossentropy: 8.0590 - val_loss: 6.6775 - val_acc: 0.5857 - val_categorical_crossentropy: 6.6775`

**Comment by bguetarni** [comment]
This can be due to: the reduction strategy that you specified (https://www.tensorflow.org/api_docs/python/tf/distribute/Strategy#reduce). It's quite tricky to determine how to fuse the results of the GPUs sometimes.
Sharing your code can help us figuring out.

**Comment by lyw615** [comment]
> This can be due to: the reduction strategy that you specified (https://www.tensorflow.org/api_docs/python/tf/distribute/Strategy#reduce). It's quite tricky to determine how to fuse the results of the GPUs sometimes.
> Sharing your code can help us figuring out.
`from keras.utils import multi_gpu_model
self.model=multi_gpu_model(self.model,gpus=self.gpu_counts)
self.model.compile(loss=categorical_crossentropy,
                           optimizer=Adam(lr=self.lr),
                           metrics=['acc', categorical_crossentropy])`

**Comment by bguetarni** [comment]
Which version of tensorflow/keras are you using ?
I cannot find `keras.utils.multi_gpu_model` in the documentation (https://www.tensorflow.org/api_docs/python/tf/keras/utils).

**Comment by lyw615** [comment]
Keras  version  is  2.3.1 ，with python =3.6.6， tensorflow-gpu=1.14.0   on Ubuntu  18.04

**Comment by bguetarni** [comment]
Can you re-run the training with `self.model=multi_gpu_model(self.model,gpus=self.gpu_counts, cpu_relocation=True)` please. Maybe upgrading the model on CPU can overcome the problem.

https://github.com/tensorflow/docs/blob/r1.14/site/en/api_docs/python/tf/keras/utils/multi_gpu_model.md
`cpu_relocation: A boolean value to identify whether to create the model's weights under the scope of the CPU. If the model is not defined under any preceding device scope, you can still rescue it by activating this option.`

**Comment by lyw615** [comment]
I have tried the resolution ,but there is  no improvement. The metrics such as  loss and acc  are still  same as the description  firstly

**Comment by tilakrayal** [comment]
@lyw615 ,
Can you please try to execute the code in latest stable version of tensorflow and keras form [here](https://www.tensorflow.org/install) and let us know if you are facing same issue.Thanks!
